### ISSUE TITLE
ALVA 640 Comfort drivers need updating

### ISSUE THREAD

**Issue by Pickl** [nvaccess/nvda#9652]
ALVA 640 Comfort cannot use the NVDA contracted/uncontracted UEB braille tables for translation.

It will display correctly on the braille display when the option is changed in the NVDA options menu but will not take the braille input. It has to use its inbuilt braille tables which are now out of date for UEB, can NVDA drivers be updated to support the ALVA 640 Comfort?

I have been in touch with the makers of the ALVA 640 comfort and they informed me it was an NVDA driver issue.

I am using NVDA version 2019.1.1 on Windows 10

**Comment by leonardder** [comment]
Thanks for your report.
Could you please try to disable HID keyboard input using thumb key 1 + smartpad enter key? Thumb key one is the left most humb key. Pressing these keys should at least say something about HID keyboard simulation disabled/enabled or unavailable.

**Comment by Pickl** [comment]
Hi, 
Hope I have given enough info. Please let me know if you need anymore.
There isn't a smartpad on the ALVA 640 comfort.
https://uk.optelec.com/products/abc-640-cf-alva-comfort-640.html

**Comment by leonardder** [comment]
In that case, there doesn't seem to be a way for you to enable or 
disable this with a gesture. In NVDA 2019.2, there will be a way to 
enable or disable this feature using NVDA's settings dialog.


For now, please try the following:


1. Open the NVDA python console with nvda+ctrl+z

2. Paste in the following and press enter:

braille.handler.display.hidKeyboardInput = False


After that, are you able to use your braille input as expected?

**Comment by Pickl** [comment]
The braille input is now working as expected. 

However, now it grade 2 the ALVA 640 comfort is now emitting a loud piercing beep after you type a word and then press space. In grade 1 it is emitting the beep after every letter.

I have tried turning NVDA and the display on and off again and this reset it to how it was before I added that into the console. I entered the code again but I am still getting the beeping.

**Comment by DrSooom** [comment]
@Pickl: It should be possible to completely disable every sound feedback via the system menu of the ALVA 640 Comfort. If I'm not wrong, you have to press B1, B3, B4 and BSpace together to open the menu. Then select "Settings" and scroll down to the different sound options.

But it would be interesting why the ALVA 640 Comfort began to beep. A deeper look at the system menu couldn't be wrong.

**Comment by Pickl** [comment]
@DrSooom thank you for that suggestion. I will give that a go. I'm not a coder myself so I wouldn't know what else to have a look at. Is there anything else I could try using the NVDA Python Console? The Optelec website doesn't state the ALVA 640 Comfort works with NVDA but it is our screen reader of choice so I would like to be able to get it working.

**Comment by Adriani90** [comment]
@Pickl are you still having this issue with NVDA 2024.2?

**Comment by Pickl** [comment]
Hi,

Sorry for delayed reply. I am no longer using the ALVA so am unsure if this is still an issue.

Thank you,

Laura


Laura Picknell
Sensory Support Practitioner
Sensory and Physical Needs
Special Education Services
Children & Young People’s Directorate

Mobile: 07732603873
Email: ***@***.******@***.***>

My work days are Tuesday, Wednesday and Thursday.

Suffolk County Council, Endeavour House, 8 Russell Road, Ipswich, IP1 2BX

Sensory and Physical Service Business Support: 01473 264717
Sensory and Physical Service Email: ***@***.******@***.***>

Any questions, please call the Inclusion Support Line: 01473 265502

We welcome and value feedback about SES: fill out our feedback form here<https://forms.office.com/r/QEbFeNNPtL>

The SES offer can be found in detail on SENCo Central, including the ISM booking link: Suffolk InfoLink | SENCO Central<https://infolink.suffolk.gov.uk/kb5/suffolk/infolink/advice.page?id=Fd-ONIQX11M>

The SES CPD offer is fully revised for 2023-24. Find out more here<https://search3.openobjects.com/mediamanager/suffolk/enterprise/files/ses_cpd_offer_2023-24.pdf>

***@***.***[Logo  Description automatically generated]

[local offer email graphic (advisor version)]

**Comment by seanbudd** [comment]
Closing as stale unless more information is provided
